src/libraries/illink-trimassemblies.targets

```diff
@@ -0,0 +1,59 @@
+<Project>
+
+  <Target Name="PrepareForAssembliesTrim"
+          DependsOnTargets="SetCommonILLinkArgs">
+
+    <PropertyGroup>
+      <!-- update debug symbols -->
+      <ILLinkArgs>$(ILLinkArgs) -b true</ILLinkArgs>
+      <!-- suppress warnings with the following codes:
+           IL2009: Could not find method A in type B specified in resource C
+           IL2025: Duplicate preserve of A in B
+      -->
+      <LinkerNoWarn>IL2009;IL2025</LinkerNoWarn>
+      <!-- https://github.com/dotnet/runtime/issues/40336 - need to also suppress the following on non-Windows:
+           IL2008: Could not find type A specified in resource B
+           and on 32-bit builds:
+           IL2012: Could not find field 'Padding' in type 'System.Runtime.CompilerServices.RawArrayData'
+      -->
+      <LinkerNoWarn Condition="'$(TargetOS)' != 'windows'">$(LinkerNoWarn);IL2008</LinkerNoWarn>
```

Comment:
Are all these `NoWarn`s necessary for OOBs? Let's try to limit the NoWarns we are passing to the trimmer, so we don't miss something.

Reply:
I believe at least some of them are, since there are members in the suppressions that are specific to linux/osx, but I'll check and remove any if possible.